### PR TITLE
KCWI Frame-typing

### PR DESCRIPTION
**Pull request by kbwestfall**
This PR provides some better default frame typing for KCWI.  This is development based on some work I did reducing some of my own KCWI data.  I'm not likely to make much more progress beyond this anytime soon, so I wanted to get this in now.

Tests pass.
```=== 324 passed in 1287.85s (0:21:27) ===```

Dev-suite launched.

### CODE REVIEW

**Comment by kbwestfall** [comment]
PS - This also updates `develop` for the recent 1.8.1 tag.

File: pypeit/display/display.py [hunk 1]
```diff
@@ -510,12 +510,12 @@ def clear_canvas(cname):
     canvas.clear()
 
 
-def clear_all():
+def clear_all(allow_new=False):
     """
```
Comment (jhennawi) [inline pypeit/display/display.py:514]: can you please document allow_new

File: pypeit/par/util.py [hunk 2]
```diff
@@ -279,36 +279,44 @@ def _find_pypeit_block(lines, group):
...
+    files = glob.glob(out)
+    return sorted(files) if sort else files
     
 
-def _read_data_file_names(lines, file_check=True):
+def _read_data_file_names(lines, file_check=True, sort=True):
     """
     Read the raw data file format
 
     Args:
```
Comment (jhennawi) [inline pypeit/par/util.py:310]: sorted --> sort in the docs.

**Review by jhennawi — CHANGES_REQUESTED** [review]
Looks good @kbwestfall. Minor documentation request and then you need to run the tests and dev suite, but otherwise consider it approved.

File: pypeit/display/display.py (hunk 1 above)
Comment (kbwestfall) [reply pypeit/display/display.py:514]: Done.

File: pypeit/par/util.py (hunk 2 above)
Comment (kbwestfall) [reply pypeit/par/util.py:310]: Good catch!  Done.

**Review by rcooke-ast — APPROVED** [review]
Seems all good to me!

The one thing I'm cautious about is that there was a period (up until Dec 2019) when the frame typing was made more complicated by arcs being turned on with the shutter closed (e.g. a flat was being taken and the arc was leaking into the flat). I don't think the changes capture this problem with "historic" datasets, so it might be worthwhile adding a comment to be aware of this for folks reducing their old KCWI data.

Thanks for the PR!!

**Comment by kbwestfall** [comment]
1 dev-suite failure.  Investigating:
```
--- PYPEIT DEVELOPMENT SUITE FAILED 1/135 TESTS (Masters ignored) ---
Failed tests:
    gemini_gnirs/32_SB_SXD pypeit_coadd_1dspec
Skipped tests:
    gemini_gnirs/32_SB_SXD pypeit_coadd_2dspec
    gemini_gnirs/32_SB_SXD pypeit_tellfit
Testing Started at 2022-02-28T22:18:35.096075
Testing Completed at 2022-03-01T07:33:32.558804
Total Time: 9:14:57.462729
```

**Comment by kbwestfall** [comment]
> The one thing I'm cautious about is that there was a period (up until Dec 2019) when the frame typing was made more complicated by arcs being turned on with the shutter closed (e.g. a flat was being taken and the arc was leaking into the flat). I don't think the changes capture this problem with "historic" datasets, so it might be worthwhile adding a comment to be aware of this for folks reducing their old KCWI data.

My data was from 2018, so I saw the issue with, e.g., the internal cals being on during the dome flat exposures, even if their shutters were closed.  The code is currently strict concerning these cases, requiring the user to type files like these themselves.  I tested the file typing against the data in the dev-suite, and the new code works fine.  I'm happy to test more data if you point me to it.

**Comment by rcooke-ast** [comment]
Great! That sounds good to me then!!

**Comment by kbwestfall** [comment]
The fix required a change to the object name.  See https://github.com/pypeit/PypeIt-development-suite/pull/190.  Afterburn scripts for gnirs now pass:
```
% ./pypeit_test afterburn -i gemini_gnirs -s 32_SB_SXD
Running tests on the following instruments:
    gemini_gnirs

Reducing data from gemini_gnirs for the following setups:
    32_SB_SXD

 0 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_sensfunc
 1 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_sensfunc
 1 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_flux_setup
 2 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_flux_setup
 2 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_flux
 3 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_flux
 3 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_coadd_1dspec
 4 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_coadd_1dspec
 4 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_coadd_2dspec
 5 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_coadd_2dspec
 5 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_tellfit
 6 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_tellfit

--- PYPEIT DEVELOPMENT SUITE PASSED 6/6 TESTS (Masters ignored) ---
Testing Started at 2022-03-01T10:07:12.560313
Testing Completed at 2022-03-01T10:16:53.520771
Total Time: 0:09:40.960458
```

**Review by jhennawi — APPROVED** [review]
Thanks!

**Pull request merged**